### DISCUSSION TITLE
Allow import i18n-t tag as es module

### DISCUSSION THREAD

**Discussion by santiagoarizti** [intlify/vue-i18n#872] [Ideas]
Hi, thanks for the awesome project.

I am using the v9 of vue-i18n with vue3

I am having issues with my IDE (phpstorm) recognizing the component `i18n-t` in my vue SFCs (single file components)

It would be of great help for the IDE (and therefore for the developer experience), to be able to import the component along the `useI18n` composable. This would greatly improve autocomplete and intellisense in the ide.

This is how I would import the component:

```
<script>
// the second import here is the component
import {useI18n, I18nT} from 'vue-i18n';

export default {
  components: {
    'i18n-t': I18nT, // note that in <script setup> this line is not necessary
  },
}
</script>
<template>
  <i18n-t keypath="hello">
    <strong>world</strong>
  </i18n-t>
</template>
```

This works fine today without need for the import, but my ide marks this as a warning 

![image](https://user-images.githubusercontent.com/8507324/148822007-24d5b60d-6130-4f8d-afa5-ff86c9afee5e.png)

I could add this tag as "ignored tags" in phpstorm

![image](https://user-images.githubusercontent.com/8507324/148821922-6d7ad634-e06a-4b9c-a4e8-4c9dad0e3f4c.png)

But I don't get any intellisense/help from the IDE about props for the tag `i18n-t`

What are your thoughts? is there any other way to make the IDE help me with this tag?

**Comment by santiagoarizti** [comment]
ok, I found a simple solution, I implemented "@web-types"

this is the `web-types.json` file as I left it

```
{
  "$schema": "https://raw.githubusercontent.com/JetBrains/web-types/master/schema/web-types.json",
  "framework": "vue",
  "name": "vue-i18n",
  "version": "9.1.9",
  "contributions": {
    "html": {
      "types-syntax": "typescript",
      "description-markup": "markdown",
      "tags": [
        {
          "name": "I18nT",
          "description": "built-in component from vue-i18n",
          "doc-url": "https://vue-i18n.intlify.dev/",
          "attributes": [
            {
              "name": "plural",
              "description": "Determines if use count 0, 1 or many",
              "default": "'0'",
              "value": {
                "kind": "expression",
                "type": "string"
              }
            },
            {
              "name": "keypath",
              "description": "The ident of the lang to be used",
              "default": "'santi'",
              "value": {
                "kind": "expression",
                "type": "string"
              }
            }
          ],
          "slots": [
            {
              "name": "default",
              "description": "interpolated string"
            }
          ]
        }
      ]
    }
  }
}
```

And in my `package.json` file I added this to point to my custom `web-types.json`

Note: I had to reboot phpstorm for this to work.

And according to [the official documentation](https://github.com/JetBrains/web-types), you can include this in your own npm package. I will add this as a feature request in "issues"

**Comment by santiagoarizti** [comment]
Issue created https://github.com/intlify/vue-i18n-next/issues/874

**Comment by TwT-L** [comment]
Why can you import `I18nT` from `vue-i18n` ?
I got Uncaught SyntaxError
`vue-i18n` does not provide an export named `I18nT`
